transposon/data.py

```diff
@@ -41,18 +43,43 @@ def __init__(self, gene_dataframe):
         self.lengths = self.data_frame.Length.to_numpy(copy=False)
         self.chromosomes = self.data_frame.Chromosome.to_numpy(copy=False)
         self.unique_genes = gene_dataframe.index.unique()
-
-    def write(self, filename):
-        """Write to disk."""
-
+        # unique() returns a list, there ought to be one element in the list
+        # and we just want it in string form, so I index on 0 MAGIC NUMBER below
+        self.chrom_of_the_subset = self.data_frame.Chromosome.unique()[0]
+
+    def write(self, filename, key='default'):
+        """Write a Pandaframe to disk.
+
+            filename (str): a string of the filename you want to write.
+            key (str): identifier for the group in the store, since we can
+            write multiple pandaframes to an hdf5 file, we need a key string
+            identifer to grab a specific pandaframe from the hdf5, so I set a
+            default value. I imagine during production we will use the
+            chromosome as the key for each pandaframe in the hdf5.
+        """
+        # self.data_frame is a PandaFrame that is why we can use to_hdf
+        self.data_frame.to_hdf(filename, key=key, mode='w')
         # NOTE consider for map reduce?
-        raise NotImplementedError()
 
-    def read(self, filename):
-        """Read from disk."""
+    @staticmethod
```

Comment:
you are correct that this doesn't need a `self` reference and are on the right track
this should be refactored to be a `classmethod` and then return an instance of GeneData
this is a common pattern where you can pseudo-overload the initializer